11_FFTBloom/main.cpp

```diff
@@ -0,0 +1,1192 @@
+
+		return m_device->createShader({ CPUShader.get(), opt.get(), m_readCache.get(), m_writeCache.get()});
+	}
+
+public:
+	// Yay thanks to multiple inheritance we cannot forward ctors anymore
+	FFTBloomApp(const path& _localInputCWD, const path& _localOutputCWD, const path& _sharedInputCWD, const path& _sharedOutputCWD) :
+		system::IApplicationFramework(_localInputCWD, _localOutputCWD, _sharedInputCWD, _sharedOutputCWD) {}
+
+	bool onAppInitialized(smart_refctd_ptr<ISystem>&& system) override
+	{
+		// Remember to call the base class initialization!
+		if (!device_base_t::onAppInitialized(smart_refctd_ptr(system)))
+			return false;
+		if (!asset_base_t::onAppInitialized(std::move(system)))
+			return false;
+
+		// Setup semaphores
+		m_timeline = m_device->createSemaphore(semaphorValue);
+		// We can't use the same sepahore for uploads so we signal a different semaphore if we need to
+		m_scratchSemaphore = m_device->createSemaphore(0);
+
+		// Get compute queue
+		m_queue = getComputeQueue();
+		uint32_t queueFamilyIndex = m_queue->getFamilyIndex();
+
+		// Create a resettable command buffer
+		{
+			smart_refctd_ptr<nbl::video::IGPUCommandPool> cmdpool = m_device->createCommandPool(queueFamilyIndex, IGPUCommandPool::CREATE_FLAGS::RESET_COMMAND_BUFFER_BIT);
+			if (!cmdpool->createCommandBuffers(IGPUCommandPool::BUFFER_LEVEL::PRIMARY, 1u, &m_computeCmdBuf))
+				return logFail("Failed to create Command Buffers!\n");
+		}
+
+		// want to capture the image data upload as well
+		//m_api->startCapture();
+
+		// Load source and kernel images
+		{
+			IAssetLoader::SAssetLoadParams lp = {};
+			lp.logger = m_logger.get();
+			lp.workingDirectory = ""; // virtual root
+			auto srcImageBundle = m_assetMgr->getAsset("../../media/colorexr.exr", lp);
+			auto kerImageBundle = m_assetMgr->getAsset("../../media/kernels/physical_flare_256.exr", lp);
+			const auto srcImages = srcImageBundle.getContents();
+			const auto kerImages = kerImageBundle.getContents();
+			if (srcImages.empty() or kerImages.empty())
+				return logFail("Could not load image or kernel!");
+			auto srcImageCPU = IAsset::castDown<ICPUImage>(srcImages[0]);
+			auto kerImageCPU = IAsset::castDown<ICPUImage>(kerImages[0]);
+			const auto srcImageFormat = srcImageCPU->getCreationParameters().format;
+			const auto kerImageFormat = kerImageCPU->getCreationParameters().format;
+
+			// Create views for these images
+			ICPUImageView::SCreationParams viewParams[2] =
+			{
+				{
+					.flags = ICPUImageView::E_CREATE_FLAGS::ECF_NONE,
+					.image = std::move(srcImageCPU),
+					.viewType = IImageView<ICPUImage>::E_TYPE::ET_2D,
+					.format = srcImageFormat,
+					.subresourceRange = {
+						.aspectMask = IImage::E_ASPECT_FLAGS::EAF_COLOR_BIT,
+						.baseMipLevel = 0u,
+						.levelCount = ICPUImageView::remaining_mip_levels,
+						.baseArrayLayer = 0u,
+						.layerCount = ICPUImageView::remaining_array_layers
+					}
```

Comment:
this is the default value for `subresourceRange`, just don't initialize it explicitly